src/NuGet.Clients/NuGet.SolutionRestoreManager/SolutionRestoreWorker.cs

```diff
@@ -154,7 +154,10 @@ public void Dispose()
             {
                 await NuGetUIThreadHelper.JoinableTaskFactory.SwitchToMainThreadAsync();
 
-                _solutionEvents.AfterClosing -= SolutionEvents_AfterClosing;
+                if (_solutionEvents != null)
```

Comment:
How does it change the event handler? It was part of Scott's PR so i just pulled it over as is and i don't see anything wrong with it since it's just an additional null check. Although it should never be null.